packages/shell-api/src/mongo.ts

```diff
@@ -264,14 +268,16 @@ export default class Mongo extends ShellApiClass {
   }
 
   @returnType('Database')
-  getDB(db: string): Database {
+  getDB<K extends StringKey<M>>(db: K): Database<M, M[K]> {
     assertArgsDefinedType([db], ['string'], 'Mongo.getDB');
     this._instanceState.messageBus.emit('mongosh:getDB', { db });
     return this._getDb(db);
   }
 
   @returnType('Collection')
-  getCollection(name: string): Collection {
+  getCollection<KD extends StringKey<M>, KC extends StringKey<M[KD]>>(
+    name: `${KD}.${KC}`
+  ): Collection<M, M[KD], M[KD][KC]> {
```

Comment:
@kmruiz Wanted to highlight this, doubt many people will actually use `.getCollection()` on a Mongo object ... but this is interesting, TS is not able to autocomplete the `name` argument for me (which, like, fair enough), but if I pass a valid `<db>.<collection>` namespace, it will return a collection with the correct schema, kind of interesting that it only partially works

Reply:
That feels like a bug right? If I recall, it should be able to autocomplete on string templates. But yeah everyone will use db.`collection` as you said, I don't see any reason now for not suggesting the usage of collections using property access instead of getCollection.

Reply:
I think it's struggling a bit with the fact that the second part (collection name) after the `.` depends on the first part (database name)

Reply:
Makes sense, this seems like a hard problem to solve for a compiler to be fair.